os/fs/smartfs/smartfs_smart.c

```diff
@@ -949,9 +1006,55 @@ static ssize_t smartfs_write(FAR struct file *filep, const char *buffer, size_t
 #endif							/* CONFIG_SMARTFS_USE_SECTOR_BUFFER */
 	}
 
+#ifdef CONFIG_SMARTFS_USE_SECTOR_BUFFER
```

Comment:
Do not write entry here, entry should be written after data sector's sync finished.
This will write entry for every write call..
Below is very simple copy operation, in this time this will
```
uint8_t buf = mallioc(st.st_blksize);
for (int i = 0; i < 16; i++) {
    read(fd1, buf, sizeof(buf);
    write(fd2, buf, sizeof(buf); //append data
}
close(fd1);
close(fd2);
```
In above case it will write entry 16times.

Reply:
This has not been updated yet.
Pending changes

Reply:
Updated